geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/KafkaGeoMessageFormatter.scala

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.locationtech.geomesa.kafka
+
+import java.io.PrintStream
+import java.nio.charset.StandardCharsets
+import java.util.Properties
+
+import com.typesafe.scalalogging.slf4j.Logging
+import joptsimple.{OptionParser, OptionSet}
+import kafka.tools.{ConsoleConsumer, MessageFormatter}
+import kafka.utils.CommandLineUtils
+import org.locationtech.geomesa.utils.geotools.SimpleFeatureTypes
+import org.opengis.feature.simple.SimpleFeatureType
+
+/** A [[MessageFormatter]] that can be used with the kafka-console-consumer to format the internal log
+  * messages of the [[KafkaDataStore]]
+  *
+  * To use add arguments:
+  *   --formatter org.locationtech.geomesa.kafka.KafkaGeoMessageFormatter
+  *   --property sft.name={sftName}
+  *   --property sft.spec={sftSpec}
+  *
+  * In order to pass the spec via a command argument all "=" characters must be replaced by "%61".  Any
+  * "%" characters that exist prior to replacement of "=" must be replaced by "%37".
+  *
+  * @see KafkaDataStoreLogViewer for an alternative to kafka-console-consumer
+  */
+class KafkaGeoMessageFormatter extends MessageFormatter {
+
+  import KafkaGeoMessageFormatter._
+
+  private var decoder: KafkaGeoMessageDecoder = null
+
+  override def init(props: Properties): Unit = {
+    if(!props.containsKey(sftNameKey)) {
+      throw new IllegalArgumentException(s"Property '$sftNameKey' is required.")
+    }
+
+    if(!props.containsKey(sftSpecKey)) {
+      throw new IllegalArgumentException(s"Property '$sftSpecKey' is required.")
+    }
```

Comment:
[minor] scala has a 'require' function that does this in a slightly cleaner manner.